### PR TITLE
Browsochrones

### PR DESCRIPTION
**Pull request by mattwigway**
Here's code integrating browsochrones for single point analysis.

### CODE REVIEW

File: lib/containers/map-control.js [hunk 1]
```diff
@@ -8,15 +8,16 @@ import {ISOCHRONE_COMPONENT} from '../constants/map'
 function mapStateToProps (state, props) {
   const {analysis, mapState, scenario} = state
   return {
-    bundleId: scenario.bundleId,
+    bundleId: scenario.currentBundle ? scenario.currentBundle.id : null,
     center: mapState.center,
     geocoderApiKey: process.env.MAPZEN_SEARCH_KEY,
     isochroneCutoff: analysis.isochroneCutoff,
-    isochroneLatLng: analysis.isochroneLatLng,
+    // TODO duplicate code below and in containers/isochrone
```
Comment (trevorgerhardt) [inline lib/containers/map-control.js]: what is this todo for? 🤔

File: lib/reducers/analysis.js [hunk 2]
```diff
@@ -16,17 +16,14 @@ export const reducers = {
   'set isochrone latlng' (state, action) {
     return {
       ...state,
-      isochroneLatLng: null
+      isochroneLatLng: action.payload
     }
   },
   'set isochrone results' (state, action) {
-    const isochrones = action.payload.avgCase.isochrones
+    let { isochrone } = action.payload
```
Comment (trevorgerhardt) [inline lib/reducers/analysis.js]: Generally use `const` whenever you can. It gives us a free check that says "this variable will never be reassigned". Not immutable, but it's enough that I'd say the two extra letters are worth it.

File: lib/utils/browsochrones.js [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+    scenarioHash: null,
+    origin: { x: NaN, y: NaN },
+    staticSiteRequest: null,
+    browsochrones
+  }
+}
+
+/** Attempt to fetch a file from the server, retrying on a 202 (Worker starting up) response */
+async function retryFetch (url, opts, { timeoutMs = RETRY_TIMEOUT_MILLISECONDS, authenticated = false }) {
+  let fetchFunc = authenticated ? authenticatedFetch : fetch
```
Comment (trevorgerhardt) [inline lib/utils/browsochrones.js]: Couldn't you always use `authenticatedFetch`? It just adds the `Authorization` header if you're logged in. But won't stop you from `fetch`ing if you're not.

https://github.com/conveyal/scenario-editor/blob/dev/lib/utils/authenticated-fetch.js#L13

File: lib/utils/browsochrones.js [hunk 4]
```diff
@@ -0,0 +1,226 @@
...
+    }, {
+      authenticated: true
+    }).then(res => res.arrayBuffer())
+
+    let { browsochrones } = instance
+
+    await Promise.all([
+      browsochrones.setQuery(metadata),
+      browsochrones.setStopTrees(stopTrees),
+      browsochrones.setTransitiveNetwork(metadata.transitive) // TODO shouldn't have to do this.
```
Comment (trevorgerhardt) [inline lib/utils/browsochrones.js]: Is this a new way of retrieving data or have we always done it this way?

File: lib/utils/browsochrones.js [hunk 5]
```diff
@@ -0,0 +1,226 @@
...
+  bikeTime: 1,
+  bikeTrafficStress: 4,
+  boardingAssumption: 'RANDOM',
+  monteCarloDraws: 220
+}
+
+// Map from scenario ID to object containing Browsochrones instance, hash of last used scenario
+// x, y coords of origin
+// TODO LRU cache
+let instances = new Map()
```
Comment (trevorgerhardt) [inline lib/utils/browsochrones.js]: how many browsochrone instances are we going to be holding here???

**Comment by trevorgerhardt** [comment]
globals isn't technically the right word there, but it fits

File: lib/utils/browsochrones.js (hunk 3 above)
Comment (mattwigway) [inline lib/utils/browsochrones.js]: It's a good point. I was thinking I also needed to use this function to grab grids from S3, but of course I don't as S3 should never need to be retried.

File: lib/utils/browsochrones.js (hunk 5 above)
Comment (mattwigway) [inline lib/utils/browsochrones.js]: Possibly 3-4

Comment (trevorgerhardt) [inline lib/utils/browsochrones.js]: Is the idea for using an LRU to limit it to 4?

Comment (mattwigway) [inline lib/utils/browsochrones.js]: Yep.

**Pull request merged**